src/modules/select/wallets/cobovault.ts

```diff
@@ -0,0 +1,261 @@
+            }
+          }
+        }
+      }
+    },
+    type: 'hardware',
+    desktop: true,
+    mobile: true,
+    osExclusions: [],
+    preferred
+  }
+}
+
+async function cobovaultProvider(options: {
+  appName: string
+  networkId: number
+  rpcUrl: string
+  BigNumber: any
+  networkName: (id: number) => string
+  resetWalletState: (options?: {
+    disconnected: boolean
+    walletName: string
+  }) => void
+}) {
+  const EthereumTx = await import('ethereumjs-tx')
+  const { default: createProvider } = await import('./providerEngine')
+
+  const BASE_PATH = "m/44'/60'/0'/0"
+
+  const { networkId, appName, rpcUrl, BigNumber, networkName } = options
+
+  const keyring = AirGapedKeyring.getEmptyKeyring()
+
+  let dPath = ''
+
+  let addressList = Array.from<string>([])
+  let enabled = false
+  let customPath = false
+
+  const provider = createProvider({
+    getAccounts: (callback: any) => {
+      getAccounts()
+        .then((res: Array<string | undefined>) => callback(null, res))
+        .catch((err: any) => callback(err, null))
+    },
+    signTransaction: (transactionData: any, callback: any) => {
+      signTransaction(transactionData)
+        .then((res: string) => callback(null, res))
+        .catch(err => callback(err, null))
+    },
+    processMessage: (messageData: any, callback: any) => {
+      signMessage(messageData)
+        .then((res: string) => callback(null, res))
+        .catch(err => callback(err, null))
+    },
+    processPersonalMessage: (messageData: any, callback: any) => {
+      signMessage(messageData)
+        .then((res: string) => callback(null, res))
+        .catch(err => callback(err, null))
+    },
+    signMessage: (messageData: any, callback: any) => {
+      signMessage(messageData)
+        .then((res: string) => callback(null, res))
+        .catch(err => callback(err, null))
+    },
+    signPersonalMessage: (messageData: any, callback: any) => {
+      signMessage(messageData)
+        .then((res: string) => callback(null, res))
+        .catch(err => callback(err, null))
+    },
+    rpcUrl
+  })
+
+  provider.setPath = setPath
+  provider.dPath = dPath
+  provider.enable = enable
+  provider.setPrimaryAccount = setPrimaryAccount
+  provider.getPrimaryAddress = getPrimaryAddress
+  provider.getAccounts = getAccounts
+  provider.getMoreAccounts = getMoreAccounts
+  provider.getBalance = getBalance
+  provider.getBalances = getBalances
+  provider.send = provider.sendAsync
+  provider.disconnect = disconnect
+  provider.isCustomPath = isCustomPath
+
+  function disconnect() {
+    dPath = ''
+    enabled = false
+    provider.stop()
+  }
+
+  async function setPath(path: string) {
+    if (path !== BASE_PATH)
+      throw new Error("CoboVault only supports standard path: m/44'/0'/0'/0/x")
+    customPath = false
+    dPath = path
+    return true
+  }
+
+  function isCustomPath() {
+    return customPath
+  }
+
+  function enable() {
```

Comment:
```suggestion
  function enable() {
    if (enabled) {
      return getAccounts()
    }
```

If a user has already connected their Cobo Vault, then adding this in will allow them switch accounts without needing to go through the connection process again.

Reply:
Thanks